### PR TITLE
fix: base64 for hashing key

### PR DESCRIPTION
**Pull request by liorzam**
New generation salt for each customer automaticlly in based64
image

for backwards compatibility old hash key still can be used, in case non of them will present an error will raise
image

## Summary by CodeRabbit

- **New Features**
  - Added a new environment variable `HASHING_KEY_SECRET_BASE64` to various Docker configuration files for enhanced security.
  - Introduced a script to generate bcrypt salts and update environment files.

- **Bug Fixes**
  - Improved error handling in `HttpExceptions.filter.ts` for specific Prisma errors.

- **Refactor**
  - Updated environment variable management in `package.json` and `env.ts` for better configuration handling.
  - Enhanced the hashing mechanism for API keys with base64 encoding.

- **Chores**
  - Added a new script to stop Docker database service before resetting and seeding in `workflows-service`.

- **Tests**
  - Updated integration tests to accommodate new salt generation and hashing mechanisms.

### CODE REVIEW

**Review by alonp99 — APPROVED** [review]

File: services/workflows-service/src/customer/api-key/utils.ts [hunk 1]
```diff
@@ -10,7 +10,9 @@ const API_KEY_LEN = 50;
 
 export const KEY_MIN_LENGTH = 5;
 
-const SALT = env.HASHING_KEY_SECRET;
+const SALT = process.env.HASHING_KEY_SECRET_BASE64
+  ? Base64.decode(process.env.HASHING_KEY_SECRET_BASE64)
+  : process.env.HASHING_KEY_SECRET;
```
Comment (Omri-Levy) [inline services/workflows-service/src/customer/api-key/utils.ts]: Can move the decoding to the zod schema's transform if you'd like.

File: services/workflows-service/src/customer/api-key/utils.ts [hunk 2]
```diff
@@ -24,7 +26,7 @@ export const hashKey = async (key: string, salt?: string) => {
       return reject(new Error('Invalid key length'));
     }
 
-    bcrypt.hash(key, salt ?? SALT, (err, hashedKey) => {
+    bcrypt.hash(key, salt ?? SALT!, (err, hashedKey) => {
```
Comment (Omri-Levy) [inline services/workflows-service/src/customer/api-key/utils.ts]: Does TypeScript warn against `null` or `undefined` due to decoding being able to fail?

File: services/workflows-service/src/env.ts [hunk 3]
```diff
@@ -11,74 +11,73 @@ const urlArrayTransformer = (value: string) => {
...
+  APP_API_URL: z.string().url().describe('The URL of the workflows-service API'),
+  COLLECTION_FLOW_URL: z.string().url().optional().describe('The URL of the Collection Flow App'),
+  WEB_UI_SDK_URL: z.string().url().optional().describe('The URL of the Web UI SDK App'),
+  DATA_MIGRATION_BUCKET_NAME: z
+    .string()
+    .optional()
+    .describe('Bucket name of Data migration folders'),
+};
+
 export const env = createEnv({
```
Comment (Omri-Levy) [inline services/workflows-service/src/env.ts:93]: Why do we need both config module and the env package?

File: services/workflows-service/src/app.module.ts [hunk 4]
```diff
@@ -43,7 +44,38 @@ import { WebhooksModule } from '@/webhooks/webhooks.module';
...
+
+  const result = envServerSchema.safeParse(config);
+
+  if (!result.success) {
+    const errors = result.error.errors.map(zodIssue => ({
+      message: zodIssue.message,
+      path: zodIssue.path.join('.'), // Backwards compatibility - Legacy code message excepts array
+    }));
+
+    throw new Error(JSON.stringify(errors, null, 2));
```
Comment (Omri-Levy) [inline services/workflows-service/src/app.module.ts:65]: Can you share a screenshot of how the error looks like in the terminal?

File: services/workflows-service/src/env.ts (hunk 3 above)
Comment (liorzam) [reply services/workflows-service/src/env.ts:93]: we dont i want to delete this one

File: services/workflows-service/src/app.module.ts (hunk 4 above)
Comment (liorzam) [reply services/workflows-service/src/app.module.ts:74]: When Slat Key isnt exists in any form(regular or base64):



When Slat Key is base64 is present in wrong format:
image

Comment (liorzam) [reply services/workflows-service/src/app.module.ts:65]: When Slat Key isnt exists in any form(regular or base64):
image


When Slat Key is base64 is present in wrong format:
image

File: services/workflows-service/src/customer/api-key/utils.ts (hunk 2 above)
Comment (liorzam) [reply services/workflows-service/src/customer/api-key/utils.ts]: validating the salt as well on startup when hashing random string
image

Comment (liorzam) [reply services/workflows-service/src/customer/api-key/utils.ts]: yeah the salt has to be present cant get undefined

**Pull request merged**